### PR TITLE
Fix uncaught exception of stof

### PR DESCRIPTION
**Pull request by gglluukk**
Fixes https://github.com/taganaka/SpeedTest/issues/60

### CODE REVIEW

**Comment by tcooper-xx** [comment]
Can confirm fix works on Ubuntu 22.04

**Comment by gglluukk** [comment]
> Can confirm fix works on Ubuntu 22.04

please show its output here

**Comment by bmxkris** [comment]
Works for me on Debian GNU/Linux 12 (bookworm)

**Comment by OboTheHobo** [comment]
Works for me as well on Fedora 39

**Comment by aidan-gibson** [comment]
Can confirm on arch

**Comment by FaazHafid** [comment]
Hi,

New here, how do I fix this with the code? Is there instructions to do that, apologies not too much familiar with GitHub or linux.

**Comment by OboTheHobo** [comment]
> Hi,
> 
> New here, how do I fix this with the code? Is there instructions to do that, apologies not too much familiar with GitHub or linux.

inside the directory for SpeedTest, run `git pull https://github.com/gglluukk/SpeedTest` and then recompile as shown in the readme (the cmake and make command)

**Comment by Stew-McD** [comment]
> Hi,
> 
> New here, how do I fix this with the code? Is there instructions to do that, apologies not too much familiar with GitHub or linux.

gh pr checkout 61

then the same install again

**Comment by ansemjo** [comment]
There's some extraneous quotes around the IP and ISP when printing the values. I'm not sure if that's due to the explicit casting to strings or if the JSON library you used does something weird?

It's not a problem for textual outputs obviously but completely breaks JSON output because that is a simple string concatenation in `main.cpp` (actual values replaced, obviously):

```json
{"client":{"ip":""x.x.x.x"","lat":"00.0000","lon":"00.0000","isp":""Deutsche Telekom AG""},"servers_online":"10","server":{"name":"Strasbourg","sponsor":"bvpn.eu","distance":"222.986","latency":"23","host":"ookla.de.eu.waselpro.com:8080"},"ping":"23","jitter":"0","_":"only latency requested"}
```

File: SpeedTest.cpp [hunk 1]
```diff
@@ -344,10 +344,14 @@ std::map<std::string, std::string> SpeedTest::parseJSON(const std::string &data)
 
     obj = json::JSON::Load(data);
     try {
-        map["ip_address"] = (std::ostringstream() << obj["ip"]).str();
-        map["isp"] = (std::ostringstream() << obj["company"]["name"]).str();
-        map["lat"] = (std::ostringstream() << obj["location"]["latitude"]).str();
-        map["lon"] = (std::ostringstream() << obj["location"]["longitude"]).str();
+        map["ip_address"] = static_cast<std::ostringstream &&>
```
Comment (ansemjo) [inline SpeedTest.cpp:343]: Try this instead to correctly dump the strings without extra quotes:

```cpp
    try {
        map["ip_address"] = obj["ip"].ToString();
        map["isp"] = obj["company"]["name"].ToString();
        map["lat"] = obj["location"]["latitude"].dump();
        map["lon"] = obj["location"]["longitude"].dump();
    } catch(...) {}
```

Comment (gglluukk) [reply SpeedTest.cpp]: I have never tried to output to JSON, so your patch corrects it as well as brings to the code more readability. 
Thanks for the fix!

**Comment by taganaka** [comment]
Thanks for helping on this!

**Pull request merged**